Neos.Flow/Classes/Mvc/Routing/Dto/RouteTags.php

```diff
@@ -0,0 +1,145 @@
+<?php
+namespace Neos\Flow\Mvc\Routing\Dto;
+
+/*
+ * This file is part of the Neos.Flow package.
+ *
+ * (c) Contributors of the Neos Project - www.neos.io
+ *
+ * This package is Open Source Software. For the full copyright and license
+ * information, please view the LICENSE file which was distributed with this
+ * source code.
+ */
+
+use Neos\Flow\Annotations as Flow;
+
+/**
+ * RouteTags to be associated with matched/resolved routes
+ *
+ * RouteTags can be added by Route Part handlers via the ResolveResult/MatchResult return values
+ * The tags will be added to the corresponding cache entries, so that they can be flushed selectively using the RouterCachingService
+ *
+ * @Flow\Proxy(false)
+ */
+final class RouteTags
+{
+
+    /**
+     * Pattern a tag must match. @see \Neos\Cache\Frontend\FrontendInterface::PATTERN_TAG
+     */
+    const PATTERN_TAG = '/^[a-zA-Z0-9_%\-&]{1,250}$/';
```

Comment:
Why duplicated this ? Why not a small object responsible to validate a tag ?

Reply:
Well. Tags may have different requirements, this one here could be less strict than the one from the cache. Then again, if the tags here are to be passed to caching, why not fully use the validation from there?

Reply:
Yes, we could just use the pattern from `\Neos\Cache\Frontend\FrontendInterface` but I wanted to avoid the coupling to another package..
We do store tags in the cache (by default) so they have to be compatible (as in: the pattern needs to be at least as strict). But that's not the only usecase. So unless we introduce a Tag ValueObject in the framework (which would be a good idea IMO) I'd rather duplicate that line of code